Python3 Compatibility

bwrsandman opened the pull request:
I am interested in using this module, but the setup.py was broken.
I also added python3 compatibility.

I will follow up with a test automation configuation for using the travis-ci.org github hook to avoid regressions on setup, python2 and python3.

>  Fix setup.py's bad closing brace 
> Fix Module imports
> Replace deprecated dict.has_key call
> Use print functions
> Alias xrange for range in python3 where they do the same thing
> Alias basestring for python3 which combines unicode and str into str
> Split __cmp__ into __lt__ and __eq__because __cmp\__ is deprecated in python3
> Replace operations.div by operations.truediv because the prior is deprecated in python3
> Add requirements file for pip install -r requirements.txt which is pretty standard for python projects.
> Add language classifier information
> Set zip_safe=False to avoid errors while installing

glenfletcher commented on Equation/core.py:
```diff
@@ -322,11 +328,17 @@ def __repr__(self):
+            return repr(self) == repr(other)
         else:
-            raise TypeError("{0:s} is not an {1:s} Object, and can't be compared to an Expression Object".format(repr(other),type(other)))
+            raise TypeError("{0:s} is not an {1:s} Object, and can't be compared to an Expression Object".format(repr(other), type(other)))
```
Why Change **cmp** to **lt** and **eq**?

bwrsandman replied:
`__cmp__` has been removed in python 3 because it forced slower comparisons.
`__lt__` and `__eq__` logically provide the same thing and since not all objects need to be sorted (you may want to only check equality), `__lt__` is optional.

More info:
https://mail.python.org/pipermail/python-3000/2008-January/011639.html

bwrsandman committed:
Fix python3 compatibility

Fix Module imports
Replace deprecated dict.has_key call
Use print functions
Alias xrange for range in python3 where they do the same thing
Alias basestring for python3 which combines unicode and str into str
Split __cmp__ into __lt__ and __eq__because __cmp__ is deprecated in python3
Replace operations.div by operations.truediv because the prior is deprecated in python3
Add requirements file for pip install -r requirements.txt which is pretty standard for python projects.
Add language classifier information
Set zip_safe=False to avoid errors while installing

bwrsandman commented:
Thank you for merging!
